module/VuFind/src/VuFind/View/Helper/Root/Record.php

```
@@ -707,4 +706,33 @@ protected function deduplicateLinks($links)
     {
         return array_values(array_unique($links, SORT_REGULAR));
     }
+
+    /**
+     * Get the source identifier + unique id of the record without spaces
+     *
+     * @param string $idPrefix Prefix for HTML ids
+     *
+     * @return string
+     */
+    public function getUniqueHtmlElementId($idPrefix = "")
+    {
+        return preg_replace(
+            "/\s+/",
+            "_",
+            ($idPrefix ? $idPrefix . '-' : '') . $this->getUniqueIdWithSourcePrefix()
+        );
+    }
+
+    /**
+     * Get the source identifier + unique id of the record
+     *
+     * @return string
+     */
+    public function getUniqueIdWithSourcePrefix()
+    {
+        if ($this->driver) {
```

This function declares that it returns a string, but it can potentially complete without returning a value at all. Maybe it should throw an exception if `$this->driver` is undefined.